plugin/goto_diagnostic.py

```diff
@@ -32,6 +35,9 @@
 import sublime_plugin
 
 
+SessionIndex = int
+SelectedIndex = int
```

Comment:
Hm, do you really want to add type aliases for such basic types like int? I would think that it makes the code more difficult to follow, rather than easier, because when you see SelectedIndex you might not remember whether it's a special class, or or a TypedDict or something else. Then you first need to look up the definition to see what it is.
And where would we stop with it; should every int and str get a special name to describe what it is, for example `session_name: SessionName` etc. ...

I'm aware that URI and DocumentUri exist as aliases for str in the protocol, but they have some kind of special meaning, namely it guarantees that the contents of those strings can be parsed as a valid URI.

Reply:
Maybe then we should add docstrings to type stubs used by LSP?

Compare (missing) info from the type stubs in this repository

![lsp](https://github.com/sublimelsp/LSP/assets/6579999/1a6c433f-820b-47f1-9b2c-eca89257e662)

with the stubs shipped by LSP-pyright:

![pyright](https://github.com/sublimelsp/LSP/assets/6579999/125c1858-b188-447f-ac81-82e948221418)

Otherwise, where would we stop? Should all and every int used in LSP get a type alias, or do we need to define certain rules for that beforehand...?

Reply:
Instead of storing the session index as an int, I would probably recommend to refactor that code and use the session name instead. This code `session = self.sessions[i]` which is used in the DiagnosticInputHandler class looks quite fragile to me (is it safe to still work correctly even when one of the sessions is closed or a new server is started in the meanwhile?)